src/rbd.cc

```diff
@@ -3586,6 +3589,10 @@ if (!set_conf_param(v, p1, p2, p3)) { \
     break;
 
   case OPT_CLONE:
+    if (!imgname || !snapname) {
```

Comment:
Now that you check for an imgname starting at line 3330, these individual checks for a populated imgname can be removed from each command block.